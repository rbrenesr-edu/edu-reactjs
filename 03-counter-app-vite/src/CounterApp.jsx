import { useState } from 'react';
import PropTypes from 'prop-types'

/*function handleAdd( event, newValue ){
    console.log(newValue);
}*/



export const CounterApp = ( {value} ) => {

    const [ counter, setCounter ] = useState(value);

    const handleAdd = ( event, newValue ) => {
        if(newValue == 'suma')
            setCounter( counter + 1 );
        if(newValue == 'resta')
            setCounter( counter - 1 );
        if(newValue == 'reset')
            setCounter( value ); 
    }

    return (
        <>     
            <h1> CounterApp </h1>
            <h2> { counter } </h2>    
              
            <button onClick={ (event) => handleAdd(event, 'suma') } > + 1 </button>
            <button onClick={ (event) => handleAdd(event, 'resta') } > - 1 </button>
            <button onClick={ (event) => handleAdd(event, 'reset') } > Reset </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,    
}
CounterApp.defaultProps = {
    value: 0,    
}