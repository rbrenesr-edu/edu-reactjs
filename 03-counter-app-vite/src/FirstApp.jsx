import PropTypes from 'prop-types'

// export function App(){
//     return (<h1>Hola mundo!!</h1>);
// }


/*Se recomienda que las contantes que no cambien de forma frecuente
se declaren fuera del componetFuntion*/
const newMessage = {
    message: 'Hola mundo',
    title:'Rafael'
};

const getNombre = ( newMessage ) => {
    return newMessage.title;
};


export const FirstApp = ( {title, subTitle, propTest} ) => {
    return (
        <> 
        <h1 data-testid="test-title">{ title }</h1>
        <h2>{ subTitle }</h2>
        <h2>{ subTitle }</h2>
        <h3>{ propTest }</h3>
        </>
    );
}
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    //subTitle: 'No hay subTitle',
    title: 'No hay titulo',
}