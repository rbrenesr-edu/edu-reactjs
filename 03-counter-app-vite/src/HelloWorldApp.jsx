// export function App(){
//     return (<h1>Hola mundo!!</h1>);
// }


// export const HelloWorldApp = () => {
//     return (
//         <div>Hello world App</div>
//     );
// }

export const HelloWorldApp = () => (<div>Hello world App!</div>);
