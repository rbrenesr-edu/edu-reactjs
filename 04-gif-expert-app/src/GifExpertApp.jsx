//t8Cg1VEQEJd1zTm4w67bKe3oE6KKRf5k
import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

  //const [categories, setCategories] = useState( [ '' ] )
  const [categories, setCategories] = useState( [] )

  const onAddCategory = ( newCategory ) =>{
    //categories.push('Programming');
    
    /**
     * no se recomienda hacer el push, poruqe no debemos mutar el 
     * objeto
     * 
     * formar de hacer el insert */
     //setCategories(['Hola mundo']);
     //setCategories([...categories ,'Hola']);

    /**Validar si el nuw category ya esta en el arreglo de categorias */

    if( categories.includes( newCategory ) ) return;

     setCategories( cat =>  [newCategory, ...cat]);
     console.log(newCategory);
  }
  

  return (
   <>

   {/* title */}
    <h1>GifExpertApp</h1>

    {/* input */}    
    <AddCategory 
      //setCategories = { setCategories }
      
      //onNewValue= { (event) => onAddCategory(event) }
      onNewValue= { onAddCategory }
    />     

    <button onClick={ () => onAddCategory( 'test' ) } > Agregar </button>

    {/* Listado de Gif */}    
    <ol>
      {
        categories.map (category  => 
        {

          return (

            <div key={ category }>                
              <li>{ category } </li>
            </div>
          )

        })
      }  
    </ol>
      {/* Gif items */}

   </>
  )
}
