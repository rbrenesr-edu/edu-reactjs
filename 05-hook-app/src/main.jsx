import React from 'react'
import ReactDOM from 'react-dom/client'
import { BrowserRouter } from 'react-router-dom'
//import { CallbackHook } from './06-memos/CallbackHook'
//import { Padre } from './07-tarea-memo/07-tarea-memo/Padre'
//import { MemoHook } from './06-memos/MemoHook'
//import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
//import { FocusScreen } from './04-useRef/FocusScreen'
//import { Layout } from './05-useLayoutEffect/Layout'
//import { Memoriza } from './06-memos/Memoriza'
//import { SimpleForm } from './02-useEffect/SimpleForm'
//import { CounterApp } from './01-useState/CounterApp'
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
//import { HooksApp } from './HooksApp'
import './index.css'
import './08-useReducer/intro-reducer'
import { TodoApp } from './08-useReducer/TodoApp'
import { MainApp } from './09-useContext/MainApp'


ReactDOM.createRoot(document.getElementById('root')).render(
    <BrowserRouter>
        {/* <MainApp /> */}
       <TodoApp/> 
    </BrowserRouter>
)
