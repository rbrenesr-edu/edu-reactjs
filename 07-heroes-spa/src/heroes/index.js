export * from './pages'
export * from './componets'
export * from './routes/HeroesRoutes'