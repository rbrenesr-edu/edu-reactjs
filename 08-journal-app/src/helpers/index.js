export * from './loadNotes'
export * from './loadNotes'