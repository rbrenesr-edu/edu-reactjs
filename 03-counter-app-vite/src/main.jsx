import React from "react";
import ReactDOM from "react-dom/client";
import { HelloWorldApp } from "./HelloWorldApp";
import { FirstApp } from "./FirstApp";
//import { CounterApp } from "./CounterApp";
import './styles.css';

/* <HelloWorldApp/>
        <FirstApp title="My Title" subTitle="subTitle" propTest="mama"/>*/ 

ReactDOM.createRoot( document.getElementById('root') ) .render (
    <React.StrictMode>       
        {/*<CounterApp value={0} />*/}
        <FirstApp />
    </React.StrictMode>
);
