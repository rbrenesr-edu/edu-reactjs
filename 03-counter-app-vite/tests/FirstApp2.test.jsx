import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

const title = 'Title FirstApp';
const subTitle = 'subTitle';

describe('Test FirstApp', () => { 
    test('Debe hacer match con el snapshopt', () => { 

        const { container } = render( <FirstApp title={title}/> );
        expect( container ).toMatchSnapshot();        
    });

    test('Debe mostrar el mensaje Test FirstApp', () => { 
        //screen.debug();
        render( <FirstApp title={title}/> );
        //screen.debug();        
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('Debe mostrar el título en un h1', () => {         
        render( <FirstApp title={title}/> );        
        expect( screen.getByRole('heading', {level: 1} ).innerHTML ).toContain(title);
    });

    test('Debe mostrar el subtítulo por props', () => {         
        
        render( 
        <FirstApp 
            title={title}
            subTitle = { subTitle }
            /> 
        );
        
        expect(screen.getAllByText(subTitle).length ).toBe(2);
        
    });

 });