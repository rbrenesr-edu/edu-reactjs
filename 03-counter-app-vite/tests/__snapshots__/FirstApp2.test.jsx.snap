// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FirstApp Debe hacer match con el snapshopt 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Title FirstApp
  </h1>
  <h2 />
  <h2 />
  <h3 />
</div>
`;
